Add workflow provenance tracking with yProv4WFS integration

hapix opened the pull request:
### Summary

This PR integrates the `yProv4WFS` library into the process graph parser to enable workflow-level provenance tracking in `openeo-pg-parser-networkx`.

### Key Changes

- Extended `graph.py` to construct yprov-compatible provenance graphs
- Added `test_pg_provenance.py` to test provenance generation functionality
- Updated `pyproject.toml` to include a validated version of `yProv4WFS`

### Integration

This provenance generation is intended to be used by the OpenEO Python client. See the corresponding PR: [Open-EO/openeo#768](https://github.com/Open-EO/openeo/pull/768), which enables returning the provenance graph as part of the `execute()` result.

Together, these changes provide full support for traceable, reproducible OpenEO workflows.

GeraldIr commented on openeo_pg_parser_networkx/graph.py:
```diff
@@ -70,6 +82,10 @@ def __repr__(self):
     def __init__(self, pg_data: dict):
+        # Make a workflow object
+        self.workflow = Workflow('wfs1', 'Workflow 1')
+        self.workflow._engineWMS = "Openeo-LocalProcessing"
```
Does this need to be explicitly "LocalProcessing"?

hapix replied:
You're absolutely right, that line was left over from testing and shouldn't be there.

GeraldIr commented on openeo_pg_parser_networkx/graph.py:
```diff
@@ -70,6 +82,10 @@ def __repr__(self):
 class OpenEOProcessGraph:
     def __init__(self, pg_data: dict):
+        # Make a workflow object
+        self.workflow = Workflow('wfs1', 'Workflow 1')
```
I would not number these. I assume this is an arbitrary name that needs to be set? If the Workflow object has defaults, just use those, otherwise go with something that looks less "hardcoded".

GeraldIr commented on openeo_pg_parser_networkx/graph.py:
```diff
@@ -411,13 +427,102 @@ def node_callable(*args, parent_callables, named_parameters=None, **kwargs):
+                if node == self.result_node:
+                    self.workflow._status= "Ok"
+
+                    # To save the provenance
```
We can either keep this in as a comment, or add a flag for local saving to the graph object?

Either way is fine, just a suggestion.

GeraldIr reviewed (approved):
No substantive suggestions, everything LGTM, all comments are basically stylistic nitpicks so don't worry too much about them. We can always change things later as long as it works.

hapix commented:
Thanks a lot @GeraldIr 
I’ve rebased the branch on main, updated the runner image to ubuntu-latest, and bumped the version in pyproject.toml to 2025.5.1 as requested.
The changes are pushed.
let me know if anything else is needed!
